hs_core/views/resource_folder_hierarchy.py

```diff
@@ -226,6 +226,22 @@ def data_store_folder_unzip(request):
     )
 
 
+def data_store_folder_unzip_public(request, pk, pathname):
+    """
+    Public version of data_store_folder_unzip, incorporating path variables
+
+    :param request:
+    :param pk:
+    :param pathname:
+    :return HttpResponse:
+    """
+    request.POST = request.POST.copy()
+    request.POST['res_id'] = pk
+    request.POST['zip_with_rel_path'] = pathname
+

```

Comment:
@aphelionz I think for public api single action is better. Client should make explicit rest call to delete the zip file. So let's force that parameter to False. @alvacouch your thoughts?